packages/@lwc/style-compiler/src/serialize.ts

```diff
@@ -344,15 +344,20 @@ function recursiveValueParse(node: any, inVarExpression = false): Token[] {
             tokens = reduceTokens(tokens);
             // The children tokens are a combination of identifiers, text and other expressions
             // Since we are producing evaluatable javascript we need to do the right scaping:
-            const exprToken = tokens.reduce((buffer, n, index) => {
-                const isTextToken = n.type === TokenType.text;
-                const normalizedToken = isTextToken ? JSON.stringify(n.value) : n.value;
-
-                // If we have a token sequence of text + expression, we need to add the concatenation operator
-                // Example: var(--x, 0 0 2px var(--y, #fff)
-                const nextToken = isTextToken && tokens[index + 1];
-                const concatOperator =
-                    nextToken && nextToken.type === TokenType.expression ? ' + ' : '';
+            const exprToken = tokens.reduce((buffer, token, index) => {
+                const isTextToken = token.type === TokenType.text;
+                const normalizedToken = isTextToken ? JSON.stringify(token.value) : token.value;
+                const nextToken = tokens[index + 1];
+
+                // If we have a token sequence of text + expression or expression + text,
+                // we need to add the concatenation operator. Examples:
+                //   var(--x, 0 0 2px var(--y, #fff))
+                //   var(--x, var(--y, #fff) 0 0 2px)
+                const shouldAddConcatinator =
+                    token.type !== TokenType.divider &&
+                    nextToken &&
+                    nextToken.type !== TokenType.divider;
```

Comment:
The comment says we're looking for a sequence of text + expression, but this is just checking if neither token is a divider. (The fourth type being identifier.) Am I misreading this?

Reply:
I'm unsure when an identifier token would show up in this part of the AST. Can you think of an example?

I added a test case for `var(foo, bar) black` in addition to the test for `var(foo, bar) 0 0 2px` that was already added.

Reply:
I've been ruminating on this a bit more and want to confirm that my thinking is correct:

- Every `token` in `tokens` should correspond to stuff inside `var(RIGHT_HERE)`.
- The only time you'd see an identifier in `RIGHT_HERE` would be as the first arg to `var`, like `var(--foo-custom-prop)`.
- You might see a fallback value, also, as well as nesting. The possible valid permutations include:
  - `var(--foo)`
  - `var(--foo, --bar)`
  - `var(0, --foo)`
  - `var(--foo, 0)`
  - `var(--foo, var(--bar, --baz))`
  - `var(--foo, 0 var(--bar))`
  - `var(--foo, var(--bar, --baz) 0)`
  - `var(--foo, var(--bar) var(--baz))` (not 100% sure this is valid, but too lazy to check)
- I do not believe the following are valid:
  - `var(--foo 0)`
  - `var(--foo, 0 bar)`
  - `var(--foo var(--baz)`
- The above lists might not be exhaustive, but I think are representative of the ways in which `identifier`s might show up within the context of a `var` function call.
- It appears correct that an `identifier` token will only show up on its own as the entire argument.
- Therefore, an `identifier` will never be adjacent to a `text` or `expression` token.

Maybe this is worth a comment. Also, I'd be happy to change `thing.type !== TokenType.divider` to `thing.type === TokenType.text || thing.type === TokenType.expression`. More explicit but also more verbose. Thoughts?

Reply:
Your explanation makes perfect sense to me. And yeah, I think the verbose one is better, since it's more explicit.

It might also be worth a comment explaining that it's impossible for an identifier to appear here. I've seen in other parts of the code where implicitly one the types would never appear (e.g. Divider), but there was no comment confirming that, so it was a bit ambiguous.